app/views/user.py

```diff
@@ -0,0 +1,50 @@
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from django.http import Http404
+
+from ..models import User
+from ..serializers import UserSerializer
+
+
+class UserList(APIView):
+
+    def get(self, request):
+        users = User.objects.all()
+        serializer = UserSerializer(users, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        serializer = UserSerializer(data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class UserDetail(APIView):
+
+    def get_object(self, pk):
+        try:
+            return User.objects.get(user_id=pk)
+        except User.DoesNotExist:
+            raise Http404
+
+    def get(self, request, pk):
+        user = self.get_object(pk)
+        serializer = UserSerializer(user)
+        return Response(serializer.data)
+
+    def put(self, request, pk):
```

Comment:
`PATCH method is the correct choice for partially updating an existing resource and PUT should only be used if you’re replacing a resource in its entirety`. Turns out PATCH might be the correct thing here

Reply:
I think we should leave this as is though.

```HTTP PATCH

HTTP PATCH requests are to make partial update on a resource. If you see PUT requests also modify a resource entity so to make more clear – PATCH method is the correct choice for partially updating an existing resource and PUT should only be used if you’re replacing a resource in its entirety.

Please note that there are some challenges if you decide to use PATCH APIs in your application:

    Support for PATCH in browsers, servers, and web application frameworks is not universal. IE8, PHP, Tomcat, Django, and lots of other software has missing or broken support for it.
    Request payload of PATCH request is not straightforward as it is for PUT request. e.g.

    HTTP GET /users/1

    produces below response:

    {id: 1, username: 'admin', email: 'email@example.org'}

    A sample patch request to update the email will be like this:

    HTTP PATCH /users/1
    [
    { “op”: “replace”, “path”: “/email”, “value”: “new.email@example.org” }
    ]

There may be following possible operations are per the HTTP specification.

[
{ "op": "test", "path": "/a/b/c", "value": "foo" },
{ "op": "remove", "path": "/a/b/c" },
{ "op": "add", "path": "/a/b/c", "value": [ "foo", "bar" ] },
{ "op": "replace", "path": "/a/b/c", "value": 42 },
{ "op": "move", "from": "/a/b/c", "path": "/a/b/d" },
{ "op": "copy", "from": "/a/b/d", "path": "/a/b/e" }
]

PATCH method is not a replacement for the POST or PUT methods. It applies a delta (diff) rather than replacing the entire resource.

```